### PR TITLE
Gas pipe topology optimization

### PR DESCRIPTION
**Pull request by jimrojerTNO**
this is a PR for topo optimization of gas pipes and electricity cables

gas pipes now only works if you manually turn off the head loss constraint
electricity gives expected solution with gurobi but not with highs

### CODE REVIEW

File: src/rtctools_heat_network/electricity_physics_mixin.py [hunk 1]
```diff
@@ -245,6 +249,67 @@ def __electricity_cable_heat_mixing_path_constraints(self, ensemble_member):
...
+            v_loss = self.state(f"{cable}.V_loss")
+            r = parameters[f"{cable}.r"]
+            # v_loss_nom = parameters[f"{cable}.nominal_voltage_loss"]
+            v_nom = parameters[f"{cable}.nominal_voltage"]
+            c_length = parameters[f"{cable}.length"]
+
+            constraint_nominal = self.variable_nominal(v_loss)
+
+            # TODO: still have to check for proper scaling
+            try:
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/electricity_physics_mixin.py:281]: I would prefer if we instead of a try except check if the cable is in the topo maps and have an if/else structure. This to have a bit more predictable behaviour of the code

File: src/rtctools_heat_network/electricity_physics_mixin.py [hunk 2]
```diff
@@ -245,6 +249,67 @@ def __electricity_cable_heat_mixing_path_constraints(self, ensemble_member):
...
+                cable_classes = self._electricity_cable_topo_cable_class_map[cable]
+                variables = {
+                    cc.name: self.variable(var_name) for cc, var_name in cable_classes.items()
+                }
+                resistances = {cc.name: cc.resistance for cc in cable_classes}
+
+                big_m = v_nom
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/electricity_physics_mixin.py:288]: I think we should do the big_m inside the for loop to compute it specific for each cable class

File: src/rtctools_heat_network/gas_physics_mixin.py [hunk 3]
```diff
@@ -124,6 +128,54 @@ def __gas_node_heat_mixing_path_constraints(self, ensemble_member):
...
+            dh = self.state(f"{pipe}.dH")
+            r = parameters[f"{pipe}.r"]
+            # v_loss_nom = parameters[f"{cable}.nominal_voltage_loss"]
+            dh_nom = parameters[f"{pipe}.nominal_head_loss"]
+            p_length = parameters[f"{pipe}.length"]
+
+            constraint_nominal = self.variable_nominal(dh)
+
+            # TODO: tobe replaced with proper headlosses relations
+            try:
```
Comment (jimrojerTNO) [inline src/rtctools_heat_network/gas_physics_mixin.py:146]: same as for electricity cables would prefer if we use if/else structure here for more predictable bahaviour

File: src/rtctools_heat_network/electricity_physics_mixin.py (hunk 2 above)
Comment (FJanssen-TNO) [reply src/rtctools_heat_network/electricity_physics_mixin.py:288]: You actually want a big_m that can cover all cable classes, in case that the size is not selected, big_m should be bigger than de v_loss - v_loss_cable size.

File: src/rtctools_heat_network/gas_physics_mixin.py (hunk 3 above)
Comment (FJanssen-TNO) [reply src/rtctools_heat_network/gas_physics_mixin.py:146]: agreed, I will change it

File: src/rtctools_heat_network/electricity_physics_mixin.py (hunk 1 above)
Comment (FJanssen-TNO) [reply src/rtctools_heat_network/electricity_physics_mixin.py:281]: Agreed, I will change it

File: src/rtctools_heat_network/electricity_physics_mixin.py (hunk 2 above)
Comment (jimrojerTNO) [reply src/rtctools_heat_network/electricity_physics_mixin.py:288]: ah ja off course, but let's use some upper bound of the variable based on max_resistance*max_current**2

Comment (FJanssen-TNO) [reply src/rtctools_heat_network/electricity_physics_mixin.py:288]: I'll think about a better upper bound. But max_resistance*max_current**2 might not be good enough because it should also overcome the voltage difference that might occur between two (disconnected) nodes due no cable being placed (DN0).
